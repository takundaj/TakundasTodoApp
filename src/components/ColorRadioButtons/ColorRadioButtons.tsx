import { View, TouchableOpacity } from "react-native";
import React from "react";
import styles from "./colorRadioButtons.styles";

type Props = {
  groupColors: string[];
  selectedColor: number;
  setSelectedColor: React.Dispatch<React.SetStateAction<number>>;
};

const ColorRadioButtons = ({
  groupColors,
  selectedColor,
  setSelectedColor,
}: Props) => {
  return (
    <View style={styles.container} testID="buttons-container">
      {groupColors.map((color, index) => (
        <TouchableOpacity
          key={index}
          testID="button"
          style={[
            styles.colorContainer,
            {
              backgroundColor: color,
              borderWidth: selectedColor === index ? 5 : 0,
              borderColor: selectedColor === index ? "turquoise" : "black",
            },
          ]}
          onPress={() => setSelectedColor(index)}
        ></TouchableOpacity>
      ))}
    </View>
  );
};

export default ColorRadioButtons;
