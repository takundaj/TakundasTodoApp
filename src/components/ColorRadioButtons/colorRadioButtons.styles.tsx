import { StyleSheet } from "react-native";
import { size } from "../../utils/helpers";

export default StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingVertical: size(4),
    marginBottom: size(4),
    shadowOffset: {
      width: 0.5,
      height: 1,
    },
    shadowOpacity: 0.8,
    shadowRadius: 2,
  },
  colorContainer: {
    height: size(8),
    width: size(8),
    borderRadius: 50,
  },
});
