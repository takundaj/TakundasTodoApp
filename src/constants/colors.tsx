export const groupColors = {
  RED: "red",
  YELLOW: "yellow",
  BLUE: "blue",
  VIOLET: "violet",
  EMERALD: "green",
  ORANGE: "orange",
  LIGHTGREY: "lightgrey",
  BLACK: "black",
};
