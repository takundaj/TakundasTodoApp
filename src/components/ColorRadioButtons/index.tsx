import ColorRadioButtons from "./ColorRadioButtons";
export default ColorRadioButtons;
