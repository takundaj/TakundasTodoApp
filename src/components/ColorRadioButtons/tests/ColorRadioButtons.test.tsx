import { render, fireEvent } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import ColorRadioButtons from "../ColorRadioButtons";
import { groupColors } from "../../../constants/colors";

const mockProps = {
  groupColors: Object.values(groupColors),
  selectedColor: 0,
  setSelectedColor: jest.fn(),
};

test("Should render ColorRadioButtons component", () => {
  const { getByTestId } = render(<ColorRadioButtons {...mockProps} />);

  const buttonsContainer = getByTestId("buttons-container");

  expect(buttonsContainer).toBeDefined;
});

test("Should render the correct amount of buttons", () => {
  const { getAllByTestId } = render(<ColorRadioButtons {...mockProps} />);

  const buttons = getAllByTestId("button");

  expect(buttons.length).toEqual(mockProps.groupColors.length);
});

test("Selected color should change when button is pressed", () => {
  const { getAllByTestId } = render(<ColorRadioButtons {...mockProps} />);

  const buttons = getAllByTestId("button");

  fireEvent.press(buttons[0]);

  expect(mockProps.setSelectedColor).toHaveBeenCalled();
});

test("Selected color should change when button is pressed", () => {
  const tree = renderer.create(<ColorRadioButtons {...mockProps} />).toJSON();

  expect(tree).toMatchSnapshot();
});
