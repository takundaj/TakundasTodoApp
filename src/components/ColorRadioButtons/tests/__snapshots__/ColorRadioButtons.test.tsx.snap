// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selected color should change when button is pressed 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 16,
      "paddingVertical": 16,
    }
  }
  testID="buttons-container"
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "red",
        "borderColor": "turquoise",
        "borderRadius": 50,
        "borderWidth": 5,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "yellow",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "blue",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "violet",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "emerald",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "orange",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "lightgrey",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "black",
        "borderColor": "black",
        "borderRadius": 50,
        "borderWidth": 1,
        "height": 32,
        "opacity": 1,
        "width": 32,
      }
    }
    testID="button"
  />
</View>
`;
